tests/org.eclipse.swt.tests.win32/JUnit Tests/org/eclipse/swt/graphics/GCWin32Tests.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Yatta Solutions
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Yatta Solutions - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.swt.graphics;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+import java.lang.reflect.Field;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.internal.DPIUtil;
+import org.eclipse.swt.internal.SWTFontProvider;
+import org.eclipse.swt.widgets.Display;
+import org.eclipse.swt.widgets.Event;
+import org.eclipse.swt.widgets.Shell;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class GCWin32Tests {
+	private Display display;
+
+	@Before
+	public void setUp() {
+		changeAutoScaleOnRuntime(true);
+		display = Display.getDefault();
+	}
+
+	@After
+	public void tearDown() {
+		changeAutoScaleOnRuntime(false);
```

Comment:
It would be better to store the old value in `setup()` and set it back in `tearDown()` instead of assuming that `false` was the initial value.

Reply:
We want it to be falsified intentionally otherwise it will mess up with other tests. The tests which need autoscaling can turn it on in their setup.

Reply:
Of course, isolating tests is a proper goal that needs to be ensured by an according teardown implementation. What @fedejeanne means is that you assume a specific default value for `autoScaleOnRuntime`. More precisely, you assume that setting the `autoScaleOnRuntime` to `false` restores the `default` state, but you do not know whether at some point in time the default may change to `autoScaleOnRuntime=true`. There might be two ways to restore the proper value:
1. There is some central place where the default value is defined
2. You have to store the original value during `setup` and restore it during `teardown`.